test/common/network/dns_impl_test.cc

```diff
@@ -240,10 +240,32 @@ class DnsResolverImplPeer {
   DnsResolverImpl* resolver_;
 };
 
+TEST(DnsImplConstructor, SupportsCustomResolvers) {
+  Event::DispatcherImpl dispatcher;
+  char addr4str[INET_ADDRSTRLEN];
+  auto addr4 = Network::Utility::parseInternetAddressAndPort("127.0.0.1:53");
```

Comment:
Maybe pick an port other than 53, as the default resolve.conf might be set to point to localhost as you describe here.